grails-app/migrations/core/2019-01-31-new-stream-fields.groovy

```diff
@@ -0,0 +1,56 @@
+package core
+databaseChangeLog = {
+	changeSet(author: "kkn", id: "new-stream-fields-1") {
+		addColumn(tableName: "stream") {
+			column(name: "all_signed", type: "bit") {
+				constraints(nullable: "true")
+			}
+		}
+	}
+	changeSet(author: "kkn", id: "new-stream-fields-2") {
+		grailsChange {
+			change {
+				sql.execute('UPDATE stream SET all_signed = false')
```

Comment:
Should define the default value in the domain class `mapping` block => the migration diff util should be smart enough to add the default value to the `addColumn` statement.

(For an example, see `requireSignedData`):
```
	static mapping = {
                //...
		requireSignedData defaultValue: "false"
	}
```

Reply:
This:
```
package core
databaseChangeLog = {
	changeSet(author: "kkn", id: "new-stream-fields-1") {
		addColumn(tableName: "stream") {
			column(name: "auto_configure", type: "bit", defaultNullValue: "true") {
				constraints(nullable: "false")
			}
		}
	}

	changeSet(author: "kkn", id: "new-stream-fields-2") {
		addColumn(tableName: "stream") {
			column(name: "storage_days", type: "integer", defaultNullValue: 365) {
				constraints(nullable: "false")
			}
		}
	}
}
```
produces:
```
problem parsing changelog.groovy: Property 'defaultNullValue' not found on object type liquibase.change.ColumnConfig (re-run with --verbose to see the stacktrace)
2019-02-05 11:43:59,208 [localhost-startStop-1] ERROR context.GrailsContextLoaderListener  - Error initializing the application: Property 'defaultNullValue' not found on object type liquibase.change.ColumnConfig
java.lang.RuntimeException: Property 'defaultNullValue' not found on object type liquibase.change.ColumnConfig
```
Even though idea suggests defaultNullValue completion. I tried both column() and addColumn().

Reply:
Was that generated by the `db-diff` script?

Reply:
For me, the script creates the following two relevant `changeSet`s for your domain class changes:

```
	changeSet(author: "hpihkala", id: "new-stream-fields-1") {
		addColumn(tableName: "stream") {
			column(defaultValue: "true", name: "auto_configure", type: "bit") {
				constraints(nullable: "false")
			}
		}
	}
	changeSet(author: "hpihkala", id: "new-stream-fields-2") {
		addColumn(tableName: "stream") {
			column(defaultValue: "365", name: "storage_days", type: "integer") {
				constraints(nullable: "false")
			}
		}
	}
```

Additionally the diff script generates the following `changeSet`s, which look like leftovers from some previous changes:

```
	changeSet(author: "hpihkala", id: "new-stream-fields-3") {
		modifyDataType(columnName: "date_created", newDataType: "datetime", tableName: "sec_user")
	}
	changeSet(author: "hpihkala", id: "new-stream-fields-4") {
		addNotNullConstraint(columnDataType: "datetime", columnName: "date_created", tableName: "sec_user")
	}
	changeSet(author: "hpihkala", id: "new-stream-fields-5") {
		modifyDataType(columnName: "last_login", newDataType: "datetime", tableName: "sec_user")
	}
	changeSet(author: "hpihkala", id: "new-stream-fields-6") {
		addNotNullConstraint(columnDataType: "datetime", columnName: "last_login", tableName: "sec_user")
	}
```